### PR TITLE
Implement query committed chaincodes in channel

### PR DESCRIPTION
**Pull request by samuelvenzi**
- Implement query committed function
- Implement query committed unit test
- Add query committed to e2e test

Signed-off-by: Samuel Venzi <samuel.venzi@me.com>

### CODE REVIEW

File: pkg/chaincode/querycommitted.go
Comment (SamYuan1990) [inline pkg/chaincode/querycommitted.go:23]: https://github.com/hyperledger/fabric/blob/main/internal/peer/lifecycle/chaincode/querycommitted.go#L223-L227
if I am correct, we can add chaincode name a parameter? ref to fabric code as example.

Comment (SamYuan1990) [reply pkg/chaincode/querycommitted.go:23]: wait, @bestbeforetoday , we removed name as parameter at https://github.com/hyperledger/fabric-protos-go-apiv2/blob/main/peer/lifecycle/lifecycle.pb.go#L1290-L1328 ?
are we able to pass QueryChaincodeDefinitionsArgs with name in fabric-protos-go-apiv2?

Comment (SamYuan1990) [reply pkg/chaincode/querycommitted.go:23]: wait, @bestbeforetoday , we removed name as parameter at https://github.com/hyperledger/fabric-protos-go-apiv2/blob/main/peer/lifecycle/lifecycle.pb.go#L1290-L1328 ?
are we able to pass QueryChaincodeDefinitionsArgs with name in fabric-protos-go-apiv2?

Comment (samuelvenzi) [reply pkg/chaincode/querycommitted.go:23]: There are currently two options for querying committed chaincodes in Fabric. One of them queries all committed in the channel and the other filters it by chaincode name.

One uses `QueryChaincodeDefinitions` tx name and the other uses `QueryChaincodeDefinition`.

Should I proceed to add both options then? In separate functions or the same?

**Review by SamYuan1990 — APPROVED** [review]
LGTM, as comments with nit enhancement maybe blocked by proto definitions and I don't want that block this PR.

File: pkg/chaincode/querycommitted.go
Comment (SamYuan1990) [reply pkg/chaincode/querycommitted.go:23]: I suppose if possible, we should support both query all and query by name.

Comment (samuelvenzi) [reply pkg/chaincode/querycommitted.go:23]: I'll let this PR be merged then, and make another adding support to query by chaincode name.

Comment (samuelvenzi) [reply pkg/chaincode/querycommitted.go:23]: > are we able to pass QueryChaincodeDefinitionsArgs with name in fabric-protos-go-apiv2?

We can use `QueryChaincodeDefinitionArgs` with name. I can do this after this PR.

```type QueryChaincodeDefinitionArgs struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	Name string `protobuf:"bytes,1,opt,name=name,proto3" json:"name,omitempty"`
}```

Comment (SamYuan1990) [reply pkg/chaincode/querycommitted.go:23]: by default, I will merge this PR before Monday, will keep this PR open for other reviewers. :-)
it's 11 pm my local time, I am going to have a rest.

Comment (SamYuan1990) [reply pkg/chaincode/querycommitted.go:23]: @samuelvenzi , you are able to have another PR for query by name after this PR been merged.

File: test/e2e_test.go
Comment (bestbeforetoday) [inline test/e2e_test.go:276]: The commit above should have caused the chaincode to be committed on all network peers where it was installed. We could confirm this by querying installed chaincode on both org peers.

**However**, we only know for sure that the commit has completed on the (Gateway) peer where we invoked the commit since that is where we observe the successfully committed transaction in the ledger. The other peers may not yet have processed the transaction. So I think this approach of querying just the peer used for the commit is reasonable.

One thing I would change though... using `peerConnections[0]` happens to be the same peer that was used for the commit but that could change if somebody changed the code above later. I think it would be better to be more explicit about the peer being used with `n_conn1` as the connection and `org1MSP` as the identity.

**Review by bestbeforetoday — COMMENTED** [review]
Nice work. Thank you for the contribution! I've added my thoughts to the discussions in the comments but generally happy with this PR as-is. Just one change to the e2e test code that I think would be worth making - see in-line comments.

File: pkg/chaincode/querycommitted.go
Comment (samuelvenzi) [reply pkg/chaincode/querycommitted.go:23]: I personally prefer `QueryCommitted` and `QueryCommittedWithName`. I didn't realize the information was slightly different, but you're right.

File: test/e2e_test.go
Comment (samuelvenzi) [reply test/e2e_test.go:276]: Thanks for the pointer, @bestbeforetoday. Changing it now.

**Review by bestbeforetoday — APPROVED** [review]

**Pull request merged**
